src/rules.jl

```diff
@@ -248,11 +248,15 @@ _abs_deriv(x) = signbit(x) ? -one(x) : one(x)
 @define_diffrule LogExpFunctions.logmxp1(x) = :((1 - $x) / $x)
 
 # binary
-@define_diffrule LogExpFunctions.xlogy(x, y) = :(log($y)), :($x / $y)
+@define_diffrule LogExpFunctions.xlogy(x, y) = 
+    :(log($y)),
+    :(z = $x / $y; iszero($x) && !isnan($y) ? zero(z) : z)
```

Comment:
This is not consistent with the rules in https://github.com/JuliaStats/LogExpFunctions.jl/pull/54, so before merging any of these PRs we should converge.

https://github.com/JuliaStats/LogExpFunctions.jl/pull/54 currently implements:
```suggestion
    :(logy = log($y); iszero($y) ? oftype(logy, NaN) : logy),
    :(w = $x / $y; iszero($x) && !isnan($y) ? zero(w) : (iszero(y) ? oftype(w, NaN) : w)
```

Similarly, the definition below is different from the one in LogExpFunctions.

Reply:
Based on your objection in https://github.com/JuliaStats/LogExpFunctions.jl/pull/54, it seems that this PR should stay as it is and that PR should be changed, or am I misinterpreting what you wrote? 

There is still an open conversation regarding https://github.com/JuliaStats/LogExpFunctions.jl/pull/54.

Your suggestion above would take us back to commit https://github.com/JuliaDiff/DiffRules.jl/pull/85/commits/fd1902da9f7e1488da73eb6f690515ffb8d55d30, at least for line 253. Is that what you would like to do?

Reply:
I don't have made up my mind regarding `NaN` - usually they are annoying but maybe that's the right thing to do here. I think the main problem with https://github.com/JuliaStats/LogExpFunctions.jl/pull/54 currently is that tests don't pass?

Reply:
Yes the tests don't pass, presumably due to the NaN treatment not matching finite differencing. I need to dig in. But I think the question of which branches to keep is kind of independent, as the tests will have to take this into account.

I guess we may choose the versions with the lowest number of branches for now, and add them later as needed?

Reply:
Sure, that might be a good guideline.